fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rewrite/scalar/ReduceCastRule.java

```diff
@@ -132,7 +132,7 @@ public boolean checkCastTypeReduceAble(Type parent, Type child, Type grandChild)
 
         // cascaded cast cannot be reduced if middle type's size is smaller than two sides
         // e.g. cast(cast(smallint as tinyint) as int)
-        if (parentSlotSize > childSlotSize && childSlotSize < grandChildSlotSize) {
+        if (parentSlotSize > childSlotSize && childSlotSize <= grandChildSlotSize) {
```

Comment:
how about `parent >= child and child < grandChild`?

Reply:
> how about `parent >= child and child < grandChild`?

@kevincai thanks for the review. as per your suggestion we change parent >= child, but in that case it will still not work because in this case the parentSize=16(VARCHAR) and childSize=8(BIGINT), grandChild=8(DOUBLE) which is same as existing erroneous code.

Reply:
I am not suggesting change it to my code, I am just wondering if the case I mentioned is considered or not. Per the comment, it is the "middle type is smaller than two sides", but actually it should be "the middle type is smaller than either of the two sides", or something more accurate?